add smqtk and requirements

msarahan opened the pull request.

Purg commented:
yay! :+1:

Purg commented on libsvm-kitware/build.sh:
```diff
@@ -0,0 +1,7 @@
+
+make lib
+mkdir $PREFIX/lib
+mv libsvm* $PREFIX/lib/
```
Going to also have to move/copy the python/svm.py and python/svmutil.py files into the site-packages directory. Optionally, we could move the svm.h heading to the ${PREFIX}/include.

msarahan replied:
It appears these files are not in the v0.1 tag that I currently have as the git revision.  That was the latest tag I could find, aside from some sort of "backup tag" before some refactoring.  Can you tag a new release of SMQTK?  It's highly frowned on to have conda packages built against untagged code.

Also, it appears that these files are part of SMQTK, not the libsvm folder within that repo.  Should they move into libsvm, or should they be packaged with SMQTK?

Purg replied:
You are right, I do need to make a new release soon. My current context is coming from the master branch. I will try to get to that in the next day or so to make sure things are in order.

As for When I do a ``find . -name "svm.py"`` from SMQTK's source (on master) it yields ``./TPL/libsvm-3.1-custom/python/svm.py``, which is in the libSVM source tree (same place as the original libSVM repo that the custom source tree is based on (https://github.com/cjlin1/libsvm)). The same should be the case for the v0.1 tag (just checked it out and checked). On that tag there is also a reference in the ``OLD_ROOT`` sub-directory, but we can ignore everything in that (removed on master and in the coming release).

msarahan replied:
My bad - I'm not sure how I missed these.  The command failed, and I went looking for Python folders, found the one in root, and did my find command in there.  Whoops.  I found it.

Now Conda won't package this because it's trying to compile these files to .pyc files and failing somehow, but that's fine - I can work on that.  Just know that this recipe won't work right now.

Purg commented:
I have also just recently needed to roll a FLANN recipe, based on yours above, but extended to be a little more comprehensive. I could paste that here or make a PR on your branch with the changes.

msarahan commented:
Great, that's very nice.  Either way works for me.  I prefer PRs just to maintain authorship history a little better.

Purg commented:
Ok, I'll get that going equally soon.

Purg commented:
I have made a v0.2 release for SMQTK

Purg commented:
Also made a PR on your ``smqtk`` branch

msarahan commented:
All packages build fine on OSX and Linux (Ubuntu 14.04.)  @Purg - any more changes you'd like to make before I merge this?

Purg commented:
I think we should bump the smqtk version from ``v0.1`` -> ``v0.2``

msarahan commented:
Bumped.  Having trouble with a missing folder: EventContentDescriptor - this appears on my system from an older installation, I think - has it been removed in the current release?

Purg commented:
Yeah, it was basically an archive of scripts from an older project. Is a ``make install`` failing because something is still referencing it?

msarahan commented:
Yes, it is still referenced at https://github.com/Kitware/SMQTK/blob/master/python/CMakeLists.txt#L2

Purg commented:
Darn. Will fix that quickly.

On Fri, Dec 18, 2015 at 5:18 PM, Mike Sarahan <notifications@github.com>
wrote:

> Yes, it is still referenced at
> https://github.com/Kitware/SMQTK/blob/master/python/CMakeLists.txt#L2
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/conda/conda-recipes/pull/506#issuecomment-165909964>.
>



-- 
Paul Tunison
R&D Engineer
Kitware, Inc.
28 Corporate Drive
Clifton Park, NY 12065 USA

Phone: (518) 371-3971 Ext.164

Purg commented:
Tagged a new release that now installs successfully (when I tested it): v0.2.1

Let me know if I messed anything else up!

msarahan commented:
All looks good to me.  Builds fine now.  Please take one more look and make sure I haven't missed anything.

ccordoba12 commented on flann/bld.bat:
```diff
@@ -0,0 +1,18 @@
+mkdir build
+cd build
+
+cmake -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% -DBUILD_C_BINDINGS:BOOL=ON -DBUILD_PYTHON_BINDINGS:BOOL=ON -DBUILD_MATLAB_BINDINGS:BOOL=OFF %SRC_DIR%
```
Please use line breaks (`^`) here to not have a huge line :-)

ccordoba12 commented on flann/meta.yaml:
```diff
@@ -0,0 +1,21 @@
+about:
+  home: http://www.cs.ubc.ca/research/flann/
+  license: BSD
+  license_file: COPYING
```
Please add  `summary`

ccordoba12 commented on libsvm-kitware/meta.yaml:
```diff
@@ -0,0 +1,11 @@
+
+about:
+  home: https://github.com/Kitware/SMQTK/tree/master/TPL/libsvm-3.1-custom
+  license: modified BSD
```
`summary` here too

ccordoba12 commented on smqtk/bld.bat:
```diff
@@ -0,0 +1,13 @@
+mkdir build
+cd build
+
+cmake -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% %SRC_DIR%
```
Line breaks here too, please :-)

ccordoba12 commented on smqtk/build.sh:
```diff
@@ -0,0 +1,9 @@
+mkdir build
+cd build
+
+cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=${PREFIX} -DCMAKE_INSTALL_PREFIX:PATH=${PREFIX} ${SRC_DIR}
```
Line breaks

ccordoba12 commented on smqtk/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+    - scikit-learn
+    - scipy
+
+
```
Remove unnecessary blank line

ccordoba12 commented on smqtk/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+
+about:
+  home: https://github.com/Kitware/SMQTK
+  license: MIT
```
`summary`

ccordoba12 commented:
@msarahan, I suggested some simple changes to improve the recipes before merging.

I know what `flann` is, but I'd like to know what the other packages are, so please don't forget to add the summaries I asked for :-)

msarahan commented:
@ccordoba12 thanks for the helpful suggestions.  I have also added a few tests for FLANN and libsvm to exercise their functionality, and added Windows build scripts for libsvm.

@Purg is there something we can do to exercise SMQTK in the same way?  I have added an import test, but that's a pretty weak test.

Purg commented:
@msarahan There is a bash script ``run_tests.sh`` in the source root (wrapper for a ``nosetest`` call).

msarahan commented:
@Purg one more check here when you have time.  I have added tests for SMQTK, as well as one nose plugin prerequisite that was not available.  I wasn't able to directly use the script you have, because we separate the test phase from the build phase, and the source code is not available (unless you copy it, which isn't really desirable).  I hope that I have captured the important bits in run_test.py.  It runs 144 tests.

Purg commented:
Basically looks file for me. The SMQTK/run_test.py looks good to me. The tests module ends up getting installed currently, so this will dive into that it seems.

Purg commented:
bump?

@msarahan: If/When this gets merged, does that mean these will get built/hosted in the primary/default conda channel?

Purg commented:
Also, [v0.2.2 has been released.](https://github.com/Kitware/SMQTK/releases/tag/v0.2.2).

msarahan commented:
Thanks for the reminder.  I'll try to get to this very soon.

Inclusion in conda-recipes does not imply availability on the primary/default channel.  Having things there implies a higher support burden, and any new packages need to be cleared with other team members.  If you'd like to ask for that, I can put you in touch, or forward your request to them.

Purg commented:
Mostly curious is all. I'll ask around here if/when we want to move towards that end.
